src/zoo.ts

```diff
@@ -0,0 +1,242 @@
+
+//  Відділ реклами використовує цей список для розсилки новин про зоопарк і рекламні акції.
+//  "Відділ реклами": Відповідає за маркетингові та рекламні заходи.
+//  Використовує список клієнтів для розсилки новин про зоопарк і рекламні акції.
+//  "Виручка": Каса збирає дані про виручку за день. Ці дані передаються в Бухгалтерію.
+//  "Бухгалтерія": Відповідає за фінансове управління зоопарку. Розпоряджається бюджетом, включно з оплатою співробітників,
+//  закупівлею корму для тварин і обслуговуванням зоопарку. Зберігає дані про всіх співробітників, тварин і виплати.
+//  Можливість генерувати фінансові звіти.
+//  "Адміністрація": Відповідає за управління співробітниками і тваринами. Може додавати і видаляти співробітників і тварин.
+
+//  Створює сповіщення про рекламні акції та інші важливі події в зоопарку.
+
+//  "Тварини": Включає в себе інформацію про кожну тварину, таку як вид, ім'я, вік, здоров'я та інші характеристики.
+//  "Співробітники": Адміністрація може додавати і видаляти співробітників.
+//  Співробітники можуть мати різні посади та обов'язки, які слід враховувати.
+//  "Бюджет": Бухгалтерія розпоряджається бюджетом і стежить за фінансами зоопарку.
+//  Можливість вести бюджетний облік і надавати фінансові звіти.
+
+enum TicketType {
+  ADULT = 'adult',
+  CHILD = 'child',
+  FAMILY = 'family',
+}
+const ticketPrice = {
+  [TicketType.ADULT]: 50,
+  [TicketType.CHILD]: 25,
+  [TicketType.FAMILY]: 100,
+};
+enum TransactionTypeEnum {
+  DEBET = 'debet',
+  CREDIT = 'credit',
+  KASSA = 'kassa',
+  SALARY = 'salary',
+}
+type AnimalHealthType = 'sick' | 'good' | 'exelent';
+class Kassa {
+  //  "Клієнти": Дані клієнтів зберігаються у Відділу реклами. dont really want to add _addDepartment propetry here, should I?
+  private _daylyIncome: number = 0;
+  private _ticket: Ticket | null = null;
+
+  public get daylyIncome(): number {
+    return this._daylyIncome;
+  }
+  public closeDay(): number {
+    // dont know how name it in another way, we use it to close day, entire day money put to account. call that dayly.
+    const summ = this._daylyIncome;
+    this._daylyIncome = 0;
+    return summ;
+  }
+  public sellTicket(ticketType: TicketType, client: Person, clientList: Person[][]): void {
+    clientList.forEach(list => list.push(client));
+    this.issueTicket(ticketType);
+    this.increaseDaylyIncome(ticketType);
+  }
+  protected issueTicket(ticketType: TicketType): Ticket {
+    return (this._ticket = new Ticket(ticketType));
+  }
+  protected increaseDaylyIncome(ticketType: TicketType): number {
+    return (this._daylyIncome += ticketPrice[ticketType]);
+  }
+}
+
+class Ticket {
+  protected _ticketType!: TicketType;
+  protected _price!: number;
+  constructor(ticketType: TicketType) {
+    (this._price = ticketPrice[ticketType]), (this._ticketType = ticketType);
+  }
+}
+class Person {
+  private readonly _id!: number;
+  private _firstName!: string;
+  private _lastName!: string;
+  private _contacts?: string;
```

Comment:
Could you please clarify why this property is marked as optional? From my perspective, it seems like it will always be required.